stairway/src/main/java/bio/terra/stairway/impl/MdcUtils.java

```diff
@@ -0,0 +1,74 @@
+package bio.terra.stairway.impl;
+
+import bio.terra.stairway.FlightContext;
+import java.util.Map;
+import org.slf4j.MDC;
+
+/**
+ * Utility methods to make Stairway flight runnables context-aware, using mapped diagnostic context
+ * (MDC).
+ */
+class MdcUtils {
+
+  /** ID of the flight */
+  static final String FLIGHT_ID_KEY = "flightId";
+
+  /** Class of the flight */
+  static final String FLIGHT_CLASS_KEY = "flightClass";
+
+  /** Class of the flight step */
+  static final String FLIGHT_STEP_CLASS_KEY = "flightStepClass";
+
+  /** Direction of the step (START, DO, SWITCH, or UNDO) */
+  static final String FLIGHT_STEP_DIRECTION_KEY = "flightStepDirection";
+
+  /** The step's execution order */
+  static final String FLIGHT_STEP_NUMBER_KEY = "flightStepNumber";
+
+  /**
+   * Null-safe utility method for overwriting the current thread's MDC.
+   *
+   * @param context to set as MDC, if null then MDC will be cleared.
+   */
+  static void overwriteContext(Map<String, String> context) {
+    if (context == null) {
+      MDC.clear();
+    } else {
+      MDC.setContextMap(context);
```

Comment:
If there was already a context map, does this completely clear it (as opposed to merging)? 

I was just wondering if the following would be more robust/clearer:

```
MDC.clear();
if (context != null) {
  MDC.setContextMap(context);
}

Reply:
They are functionally the same per MDC's documentation, but I agree that your version is clearer.  Will update.